openstack/standalone-cinder/pkg/provisioner/vsbroker.go

```diff
@@ -31,6 +31,7 @@ type volumeServiceBroker interface {
 	disconnectCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error
 	unreserveCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error
 	deleteCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error
+	getCinderVolumeState(vs *gophercloud.ServiceClient, volumeID string) (string, error)
```

Comment:
This reminds me that we should have some tests for this functionality.  At least the positive flow, volume not found, and volume in wrong state.  We can also check that a found volume is not deleted in the event of a later error.

Reply:
Do you mean unit test for this function?

Reply:
@aglitke Thanks, I've addressed other comments. For this one, I think might need to add another provisioner and deleting test case using different volume options (useExistingVolume=true), see [here](https://github.com/kubernetes-incubator/external-storage/blob/master/openstack/standalone-cinder/pkg/provisioner/provisioner_test.go#L33), right?